### PR TITLE
LibWeb: Speed up CSS namespace checking

### PR DESCRIPTION
**Pull request by AtkinsSJ**
CSSStyleSheet now caches the CSSNamespaceRule for the default namespace, which is the only one we currently care about. This saves us from iterating over its list of rules every time we want to know what that default namespace is.

The spec dictates that `@namespace` rules are only valid near the start of a stylesheet, so we also take advantage of that to quit searching for namespaces as soon as we see a non-import rule.

Also renamed `namespace_filter()` to `default_namespace()` since that's what it actually returns.

This makes github.com/serenityos/serenity snappy again. :^)

### CODE REVIEW

**Comment by AtkinsSJ** [comment]
Changes: Add a CSSRuleList::on_change callback, and use that for recalculating the namespace cache. This means it'll be updated when the CSSRuleList is modified directly as well as when the CSSStyleSheet is.

File: Userland/Libraries/LibWeb/CSS/CSSStyleSheet.cpp [hunk 1]
```diff
@@ -138,17 +143,40 @@ void CSSStyleSheet::set_style_sheet_list(Badge<StyleSheetList>, StyleSheetList*
...
-            auto& namespace_rule = verify_cast<CSSNamespaceRule>(*rule);
-            if (!namespace_rule.namespace_uri().is_empty() && namespace_rule.prefix().is_empty())
-                return namespace_rule.namespace_uri().view();
+        // "Any @namespace rules must follow all @charset and @import rules and precede all other
+        // non-ignored at-rules and style rules in a style sheet.
+        // ...
+        // A syntactically invalid @namespace rule (whether malformed or misplaced) must be ignored."
+        // https://drafts.csswg.org/css-namespaces/#syntax
+        switch (rule->type()) {
+        case CSSRule::Type::Import:
```
Comment (stelar7) [inline Userland/Libraries/LibWeb/CSS/CSSStyleSheet.cpp:163]: Shouldnt `Charset` be added here too?

Comment (awesomekling) [reply Userland/Libraries/LibWeb/CSS/CSSStyleSheet.cpp:163]: We don't have `Charset` as a rule type, so this code should be okay for now :)

Comment (AtkinsSJ) [reply Userland/Libraries/LibWeb/CSS/CSSStyleSheet.cpp:163]: Iirc, `@charset` doesn't exist as a DOM object. Just had a quick look at the specs and that seems to be right, though they've changed their minds a couple of times.

**Pull request merged**